File: BitManipulation/LeetCode/Medium/OGArrayofPrefixXOR/OGArrayOfPrefixXOR.java
// https://leetcode.com/problems/find-the-original-array-of-prefix-xor/
public class OGArrayOfPrefixXOR {
    public int[] findArray(int[] pref) {
        int ans[] = new int[pref.length];
        ans[0] = pref[0];
        for (int i = 1; i < pref.length; i++) {
            ans[i] = pref[i] ^ pref[i - 1];
        }
        return ans;
    }
}

File: BitManipulation/LeetCode/Medium/OGArrayofPrefixXOR/OGArrayofPrefixXOR.cpp
#include<bits/stdc++.h>
using namespace std;
vector<int> findArray(vector<int> &pref)
{
    vector<int> ans(pref.size());
    ans[0] = pref[0];
    for (int i = 1; i < pref.size(); i++)
    {
        ans[i] = pref[i] ^ pref[i - 1];
    }
    return ans;
}
int main(void) {
    
    return 0;
}